src/api/eventTypes.js

```diff
@@ -159,6 +161,18 @@ export type UpdateMessageFlagsEvent = $ReadOnly<{|
   flag: empty, // TODO fill in
   all: boolean,
   messages: $ReadOnlyArray<number>,
+
+  message_details?: {|
+    [string]: {|
+      type: 'stream' | 'private',
+      mentioned?: true,
+      // type: 'stream'
+      stream_id?: number,
+      topic?: string,
+      // type: 'private'
+      user_ids?: UserId,
```

Comment:
This is best expressed as a union of two object types -- that way the relationship between `type` being e.g. `'stream'`, and `stream_id` and `topic` being definitely present, is in the type, and the code that consumes the object can get simpler. I'll demonstrate in a small added commit.